Sources/SecureEnclaveValet.swift

```diff
@@ -0,0 +1,285 @@
+        }
+    }
+    
+    /// Removes a key/object pair from the keychain.
+    /// - returns: `false` if the keychain is not accessible.
+    @objc(removeObjectForKey:)
+    @discardableResult
+    public func removeObject(for key: Key) -> Bool {
+        return execute(in: lock) {
+            switch Keychain.removeObject(for: key, options: keychainQuery) {
+            case .success:
+                return true
+                
+            case .error:
+                return false
+            }
+        }
+    }
+    
+    /// Removes all key/object pairs accessible by this Valet instance from the keychain.
+    /// - returns: `false` if the keychain is not accessible.
+    @objc
+    @discardableResult
+    public func removeAllObjects() -> Bool {
+        return execute(in: lock) {
+            switch Keychain.removeAllObjects(matching: keychainQuery) {
+            case .success:
+                return true
+                
+            case .error:
+                return false
+            }
+        }
+    }
+    
+    /// Migrates objects matching the input query into the receiving SecureEnclaveValet instance.
+    /// - parameter query: The query with which to retrieve existing keychain data via a call to SecItemCopyMatching.
+    /// - parameter removeOnCompletion: If `true`, the migrated data will be removed from the keychain if the migration succeeds.
+    /// - returns: Whether the migration succeeded or failed.
+    /// - note: The keychain is not modified if a failure occurs.
+    @objc(migrateObjectsMatchingQuery:removeOnCompletion:)
+    public func migrateObjects(matching query: [String : AnyHashable], removeOnCompletion: Bool) -> MigrationResult {
+        return execute(in: lock) {
+            return Keychain.migrateObjects(matching: query, into: keychainQuery, removeOnCompletion: removeOnCompletion)
+        }
+    }
+    
+    /// Migrates objects matching the vended keychain query into the receiving SecureEnclaveValet instance.
+    /// - parameter keychain: An objects whose vended keychain query is used to retrieve existing keychain data via a call to SecItemCopyMatching.
+    /// - parameter removeOnCompletion: If `true`, the migrated data will be removed from the keychfain if the migration succeeds.
+    /// - returns: Whether the migration succeeded or failed.
+    /// - note: The keychain is not modified if a failure occurs.
+    @objc(migrateObjectsFromKeychain:removeOnCompletion:)
+    public func migrateObjects(from keychain: KeychainQueryConvertible, removeOnCompletion: Bool) -> MigrationResult {
+        return migrateObjects(matching: keychain.keychainQuery, removeOnCompletion: removeOnCompletion)
+    }
+    
+    // MARK: Private Properties
+    
+    private let service: Service
+    private let lock = NSLock()
+    private let keychainQuery: [String : AnyHashable]
+}
+
+
+// MARK: - Objective-C Compatibility
```

Comment:
Should all the methods below have the notforswift (read: ⛔️) prefix?

Reply:
they should! Now fixed.

We had to go with 🚫 instead of ⛔️ since it looks like Xcode doesn't allow composed-character emojis in code.

For reference, 🚫's unicode representation is `U+1F6AB`, and ⛔️'s unicode representation is `U+26D4 U+FE0F`.